mtgjson5/providers/tcgplayer.py

```diff
@@ -148,6 +148,71 @@ def generate_today_price_dict(
         return dict(results)
 
 
+def get_tcgplayer_sku_data(group_id_and_name: Tuple[str, str]) -> List[Dict]:
+    """
+    finds all sku data for a given group using the TCGPlayer API
+    :param group_id_and_name: group id and name for the set to get data for
+    :return: product data including skus to be parsed into a sku map
+    """
+    magic_set_product_data = []
+    api_offset = 0
+
+    while True:
+        api_response = TCGPlayerProvider().download(
+            "https://api.tcgplayer.com/catalog/products",
+            {
+                "offset": str(api_offset),
+                "limit": 100,
+                "categoryId": 1,
+                "includeSkus": True,
+                "groupId": group_id_and_name[0],
+            },
+        )
+
+        if not api_response:
+            # No more entries
+            break
+
+        response = json.loads(api_response)
+        if not response["results"]:
+            # Something went wrong
+            break
+
+        for product_data in response["results"]:
+            magic_set_product_data.append(product_data)
+
+        api_offset += len(response["results"])
+
+    return magic_set_product_data
+
+
+def generate_tcgplayer_sku_map(tcgplayer_set_sku_data: List[Dict]) -> Dict[str, List]:
```

Comment:
Your types don't make sense here. For non-primatives, you'll need to be more detailed. List[Dict[str, str]] for example. You'll need to figure out your types.